esmvaltool/cmorizers/data/formatters/datasets/esdc.py

```diff
@@ -0,0 +1,84 @@
+"""ESMValTool CMORizer for Earth System Data Cube data.
+
+<We will add some useful info here later>
+"""
+import logging
+from pathlib import Path
+import xarray as xr
+from esmvalcore.preprocessor import monthly_statistics 
+from esmvaltool.cmorizers.data import utilities as utils
+
+logger = logging.getLogger(__name__)
+
+
+def fix_time_coordinate(cube):
+    time = cube.coord(axis = 'T')
+    time.convert_units('days since 1850-1-1 00:00:00.0')
+
+
+def fix_cube(short_name, var, cube, cfg):
+    cmor_info = cfg['cmor_table'].get_variable(var['mip'], short_name)
+    # fix_time_coordinate(cube)
+    utils.fix_coords(cube)
+    logger.info(cmor_info.units)
+    cube.convert_units(cmor_info.units)
+    if 'height2m' in cmor_info.dimensions:
+        utils.add_height2m(cube)
+    cube = monthly_statistics(cube, operator="mean") # Fix frequency
+    return cube
+
+
+def cmorization(in_dir, out_dir, cfg, cfg_user, start_date, end_date):
+    """Cmorize the dataset."""
+
+    # This is where you'll add the cmorization code
+    # 1. find the input data
+    logger.debug("in_dir: '%s'", in_dir)
+    logger.debug("cfg: '%s'", cfg)
+    logger.debug("cfg_user: '%s'", cfg_user)
+
+
+    attributes = cfg['attributes']
+    variables = cfg['variables']
+
+    version = attributes['version']
+
+    filename_pattern = cfg['filename'].format(
+        grid=attributes['grid'],
+        chunking=attributes['chunking'],
+        version=version)
+
+    def cmorize_cube(path):
+        logger.info(f'Opening zarr in "{path}"')
+        try:
+            ds = xr.open_zarr(path, consolidated=True)
+        except KeyError as e:
```

Comment:
is that a KeyError or OSError?

Reply:
There's a KeyError that happens when trying to open a zarr array instead of a zarr dataset. When using wildcards in the cmor_config Yaml file the pattern could match those as well, so I used this to ignore them.

But it could be removed and leave it to users' responsibility to be careful with wildcards.

Reply:
sorry, bud, forgot to reply here - up to you - no stroganoff preference from me :grin:

Reply:
Can I consider this resolved then?